login/service_login_blackbox_test.go

```diff
@@ -188,26 +188,41 @@ func (s *serviceLoginBlackBoxTest) runLoginEndToEnd() {
 func (s *serviceLoginBlackBoxTest) TestOauth2LoginEndToEnd() {
 	s.approved = true
 	s.alreadyLoggedIn = false
-	s.runOauth2LoginEndToEnd()
+	modeFragment := "fragment"
```

Comment:
you could use subtests here :) (seriously, named subtests would give some insight about what you're actually testing)

Reply:
I agree - but since we are avoiding that paradigm, I avoided it as well :|